pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -357,7 +470,7 @@ func (r *ReconcileMachineHealthCheck) healthCheckTargets(targets []target, timeo
 		}
 
 		if t.Machine.DeletionTimestamp == nil {
-			currentHealthy++
+			currentHealthy = append(currentHealthy, t)
 		}
 	}
 	return currentHealthy, needRemediationTargets, nextCheckTimes, errList
```

Comment:
Just to make sure I understand correctly.
The bottom line is that you recommend leaving the code as is, even though it technically can be changed.

Reply:
My suggestion would be either to omit the return values or bring back the var declaration at the beginning instead of declaring them in the function signature.

I prefer the latter but it's up to you.